### PR TITLE
Arc 1458 arc email reporting for uusa finalize staging tables needed for email mart to build for uusa

### PR DESCRIPTION
**Pull request by Frydafly**
# Pull Request for dbt-arc-functions

- [x] Create branch of dbt-arc-functions using JIRA extension so that naming convention is consistent

- [x] If you've created a macro, make sure it is documented by running the notebook: `create_docs.ipynb` util in this repo

- [x] Test branch on a development branch of an existing client (ideally the one that raised the bug). Do this by changing revision of the package to the branch name in `packages.yml` file in dbt State client below.

- [x] If new macro files were created or updated, re-run `create_or_update_standard_models.py` for the client dbt project, replacing models and dependencies
*A note that for this change, full schema delete of staging schema is required to test and to run* FYI @dmbluestate @ryantimjohn  

- [x] check in dbt that client package successfully runs `dbt deps`, `dbt compile`, and `dbt run`; screenshot this page
Screenshot 2023-05-03 at 6 35 42 PM

- [x] query the resulting staging model in bigquery (or by previewing mart in dbt cloud, or your IDE); screenshot this
Screenshot 2023-05-03 at 6 36 11 PM

- [x] If exists, link bug issue and jira tickets to PR
https://bluestate.atlassian.net/browse/ARC-1458

### CODE REVIEW

**Comment by Frydafly** [comment]
@ryantimjohn -- I'm not passing the sqlfmt linter unfortunately :( 

and just noticed I'm missing one macro documented -- will do that now! 

but this is ready for review

**Comment by Frydafly** [comment]
And the PR for running this in UUSA is here: https://github.com/bsd/arc-uusa-dbt/pull/5

**Comment by Frydafly** [comment]
hola @ryantimjohn , here's the error message I get in "check for properly formatted docs" 

Screenshot 2023-05-03 at 6 44 41 PM

**Comment by ryantimjohn** [comment]
@Frydafly, I don't think I can reasonably make a PR against this branch because of Git Conflicts resolution, will link to the place where I'll make the changes for better error message but, long and short, you're getting this error because of all the blank docs.

**Comment by ryantimjohn** [comment]
Ah... I added an error message for this but it's on the main branch (you can do an update from main in GH Desktop or git.)

I think the dbt-repo you're trying to create the docs from might not have dbt-codegen installed. OR you might need to run `dbt run` in that directory first.

**Comment by Frydafly** [comment]
@ryantimjohn what do you think if we add dbt-labs/dbt-codegen to the packages that first_run...py builds? since it's a requirement for documentation? let me know what you think and I can make ticket or issue to add that to our list.

**Comment by Frydafly** [comment]
@ryantimjohn I ran create_docs, this time with codegen installed 😵‍💫

**Comment by ryantimjohn** [comment]
> @ryantimjohn what do you think if we add dbt-labs/dbt-codegen to the packages that first_run...py builds? since it's a requirement for documentation? let me know what you think and I can make ticket or issue to add that to our list.

Great idea, added ticket here!
https://bluestate.atlassian.net/browse/ARC-1468

**Comment by ryantimjohn** [comment]
I think your linter's not passing because outdated packages locally @Frydafly , copy pasting from other PR:
You should run:

pipx install sqlfmt --upgrade

and

pipx install yamlfix --upgrade

(if you're using pipx, which automatically installs it in it's own environment (recommended) otherwise, use pip)

**Comment by Frydafly** [comment]
> I think your linter's not passing because outdated packages locally @Frydafly , copy pasting from other PR: You should run:
> 
> pipx install sqlfmt --upgrade
> 
> and
> 
> pipx install yamlfix --upgrade
> 
> (if you're using pipx, which automatically installs it in it's own environment (recommended) otherwise, use pip)

thanks @ryantimjohn -- this certainly helped with yamlfix, unfortunately still did not change any of my files in sqlfmt -- I'll look into that issue some more but it's puzzling me!

**Comment by Frydafly** [comment]
hi! all checks are passing now! @dmbluestate @ryantimjohn

File: utils/create_docs.ipynb
Comment (ryantimjohn) [inline utils/create_docs.ipynb:1]: I don't think you want to change these @Frydafly

**Review by ryantimjohn — APPROVED** [review]
Lot of changes here, but looks good to me except can you revert the change on create_docs.ipynb @Frydafly ?

File: utils/create_docs.ipynb
Comment (Frydafly) [reply utils/create_docs.ipynb:1]: Hm! I only meant to add documentation. Let me double check this

Comment (Frydafly) [reply utils/create_docs.ipynb:1]: Right! That was a mistake

**Comment by dmbluestate** [comment]
@Frydafly @ryantimjohn @lbartkowiak1 Lots of changes here. I am seeing that we deleted the UUSA-specific email_campaign tables that were tested and working, and we've replaced them with tables that are mainly showing nulls. I am also seeing that we deleted the src_staging tables that were built and were approved in a previous PR. I am trying to understand why there were deleted as we will need them to de-duplicate the src tables from stitch and also will need them to merge in historical src data tables eventually. Is there another plan for these use cases? Could you revert these changes until we can review, discuss and validate the changes in this PR that are uusa specific?

**Comment by Frydafly** [comment]
> @Frydafly @ryantimjohn @lbartkowiak1 Lots of changes here. I am seeing that we deleted the UUSA-specific email_campaign tables that were tested and working, and we've replaced them with tables that are mainly showing nulls. I am also seeing that we deleted the src_staging tables that were built and were approved in a previous PR. I am trying to understand why there were deleted as we will need them to de-duplicate the src tables from stitch and also will need them to merge in historical src data tables eventually. Is there another plan for these use cases? Could you revert these changes until we can review, discuss and validate the changes in this PR that are uusa specific?

https://github.com/bsd/dbt-arc-functions/pull/236/files 

here's my draft PR where I am reverting so that your work isn't removed, and then I'll work to consolidate a bit more

**Pull request merged**